lightly/api/swagger_rest_client.py

```diff
@@ -1,8 +1,27 @@
+import json
 from typing import Any, Dict, Optional, Tuple, Union
 
 from lightly.openapi_generated.swagger_client.api_client import Configuration
+from lightly.openapi_generated.swagger_client.exceptions import ApiException, UnauthorizedException
+from lightly.openapi_generated.swagger_client.models.api_error_code import ApiErrorCode
 from lightly.openapi_generated.swagger_client.rest import RESTClientObject
 
+class PrettyPrintApiException(ApiException):
+
+    def __init__(self, current_exception: ApiException):
+        super().__init__(current_exception.status, current_exception.reason)
+        self.body = current_exception.body
+        self.headers = current_exception.headers
+
+    def __str__(self):
+        error_message = f"\nError Code: {self.status}\n"
+        error_message += f"Error Reason: {self.reason}\n"
+        error_body_dict = json.loads(self.body)
```

Comment:
this can easily fail if its not a well formatted error code from the server as an object.

Reply:
I put additional checks and a try-catch around it.